### PR TITLE
feat: implemented data and logic split pattern

### PR DESCRIPTION
**Pull request by tyriis**

### CODE REVIEW

File: data/channels/alerts.yaml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: terraform.techtales.io/v1alpha1
+kind: DiscordTextChannel
+metadata:
+  name: alerts
+  namespace: techtales.io
+spec:
+  displayName: 🚨・alerts
+  topic: from Prometheus Alertmanager
+  syncsyncPermissionsWithCategory: false
```
Comment (jazzlyn) [inline data/channels/alerts.yaml:10]: typo (sync)

File: data/channels/botkube.yaml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+---
+apiVersion: terraform.techtales.io/v1alpha1
+kind: DiscordTextChannel
+metadata:
+  name: botkube
+  namespace: techtales.io
+spec:
+  displayName: 🤖・botkube
+  topic: monitor + act on your K8s cluster
+  syncsyncPermissionsWithCategory: false
```
Comment (jazzlyn) [inline data/channels/botkube.yaml:10]: typo (sync)

File: modules/assets/png-loader/locals.tf [hunk 3]
```diff
@@ -0,0 +1,7 @@
+locals {
+  path = "assets"
+  assets = {
+    for file in fileset("${path.root}/../${local.path}", "*.png") :
```
Comment (jazzlyn) [inline modules/assets/png-loader/locals.tf:4]: :magic_wand:

File: modules/discord/category/main.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
+terraform {
+  required_version = ">= 1.8.0"
```
Comment (jazzlyn) [inline modules/discord/category/main.tf:2]: keep tf version everywhere same (1.9.0)

File: modules/discord/category/variables.tf [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+      name      = string
+      namespace = optional(string)
+    })
+    spec = object({
+      displayName = optional(string)
+      channels    = list(string)
+    })
+  })
+
+  validation {
```
Comment (jazzlyn) [inline modules/discord/category/variables.tf:15]: :100:

File: modules/assets/png-loader/locals.tf (hunk 3 above)
Comment (tyriis) [reply modules/assets/png-loader/locals.tf:4]: we are in the path ./terraform our assets are in the path ./assets

terraform is called from ./terraform (path.root) from there we need to go back (..) and then enter the folder assets (local.path) from there we load all files matching *.png

**Review by jazzlyn — APPROVED** [review]

**Comment by tyriis** [comment]
atlantis apply

**Comment by jazzlyn** [comment]
atlantis plan

**Comment by jazzlyn** [comment]
atlantis apply

**Pull request merged**